frontend/src/metabase/components/form/CustomForm/CustomFormSubmit.tsx

```diff
@@ -65,6 +68,28 @@ CustomFormSubmitLegacyContext.contextTypes = _.pick(
   "submitTitle",
   "renderSubmit",
   "disablePristineSubmit",
+  "shouldPersistError",
+  "fields",
 );
 
+function areAllFieldsTouched(fields: NestedFormField): boolean {
+  const allFields = traverseAllFields(fields);
```

Comment:
Need to traverse nested key as form could be nested e.g. database form has fields under `details` key.

https://github.com/metabase/metabase/blob/c67b62335437d5a22542d9a6757582a4f5471bd9/frontend/src/metabase/entities/databases/forms.js#L248-L250